src/Compatibility/Microsoft.DotNet.PackageValidation/BaselinePackageValidator.cs

```diff
@@ -1,110 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System.Collections.Generic;
-using System.Linq;
-using Microsoft.DotNet.ApiCompatibility;
-using Microsoft.DotNet.ApiCompatibility.Abstractions;
-using Microsoft.DotNet.Compatibility.ErrorSuppression;
-using NuGet.ContentModel;
-using NuGet.Frameworks;
-
-namespace Microsoft.DotNet.PackageValidation
-{
-    /// <summary>
-    /// Validates that no target framework / rid support is dropped in the latest package.
-    /// Reports all the breaking changes in the latest package.
-    /// </summary>
-    public class BaselinePackageValidator
```

Comment:
Did the code on these files change at all? or was this just a file move?

Reply:
Noted in the commit and PR description but to provide more detail, all validators changed in the following way:
- They now accept an PackageValidatorOption object in the Validate method so that they can operate on multiple packages in the future. The PackageValidatorOption type contains all the information that is specific to a package validation run. The PackageValidator type itself can be cached and reused.
- The ApiCompatRunner is now instantiated during validation as with its current design, it's specific to a package validation run.
- Fix: The ApiCompatRunner's RunApiCompat is now only called when the `RunApiCompat` is true to avoid unnecessary work.

An additional fix landed in the `CompatibleTFMValidator` which compared the same assemblies twice when the runtimeAsset was the same as the runtimeRidSpecificAsset.

And last but not least as mentioned in the Team chat, I noticed that the `BaselinePackageValidator` compares baseline rid specific runtime asset against latest runtime asset which is unintentional. I didn't fix that yet though.

Reply:
I added a comment in the end touching this point, but I would prefer if the functional changes fixing these bugs happen in separate PRs, that could even optionally be tied to a specific issue calling out the problem. This makes it easier to track the fixes, review the changes, and gives you the option of reverting them easily in case those changes caused a regression.